nbgrader/converters/base.py

```diff
@@ -359,3 +363,28 @@ def _handle_failure(gd):
 
             self.log.error(msg)
             raise NbGraderException(msg)
+
+    def recursively_archive(self, directory, archive_name):
+        all_files_and_folders = [ item for item in os.listdir(directory) if item != archive_name+".zip" ]
+
+        for item in all_files_and_folders:
+            full_path = os.path.join(directory, item)
+            if os.path.isfile(full_path):
+                self.archive.write(os.path.join(self.writer.build_directory, full_path.split(archive_name, 1)[-1][1:]), archive_name+"/"+full_path.split(archive_name, 1)[-1][1:])
```

Comment:
It might be a bit cleaner to use `os.path.relpath` here rather than splitting on `archive_name`, i.e. `os.path.relpath(fullpath, archive_name)`

Reply:
I would also not rely on `self.writer.build_directory` here, but pass in the root directory that you care about as an argument to the function and then use that.